src/retrace/retrace.py

```diff
@@ -1754,16 +1756,13 @@ def prepare_debuginfo(self, vmcore, chroot=None, kernelver=None, crash_cmd=["cra
         # Note the dependency from this code on the debuginfo file list
         if vmlinux is None:
             vmlinux_debuginfo = os.path.join(debugdir_base, vmlinux_path.lstrip("/"))
+            cache_files_from_debuginfo(debuginfo, debugdir_base, [vmlinux_path])
```

Comment:
The call is prefaced by 'if vmlinux is None:'.  So if that is true, we don't have a vmlinux file yet in the cache (see prior code that looks in the standard location for it).  Then even if cache_files_from_debuginfo is called, we still need to check if the file exists (because it's possible cache_files_from_vmlinux may silently fail).  In short, yes is it necessary and no it is not unconditionally called.  I hope that explains it.  If you're thinking of something else and I missed what you're saying please clarify maybe with further code suggestion.  Thanks.